### PR TITLE
Add restrictChatMember API

### PR DESCRIPTION
**Pull request by Hentioe**
I passed the test with my own bot, please merge it, thank you.

### CODE REVIEW

**Comment by zhyu** [comment]
Thanks for the m-r!

It seems the travis CI integration was not working well, I triggered a re-build, the build status is now available.
You can see the build failed, please fix it.

Could you please add tests for this new API as well?

Thanks again!

File: lib/nadia/model.ex [hunk 1]
```diff
@@ -367,6 +367,44 @@ defmodule Nadia.Model do
...
+            can_add_web_page_previews: boolean(),
+            can_change_info: boolean(),
+            can_invite_users: boolean(),
+            can_pin_messages: boolean()
+          }
+  end
+
+  defimpl String.Chars, for: ChatPermissions do
+    def to_string(value) do
+      empty_keys =
```
Comment (zhyu) [inline lib/nadia/model.ex:395]: to drop keys with `nil` value, you can do this:
```elixir
value
|> Map.from_struct()
|> Enum.filter(fn {_, v} -> v != nil end)
|> Enum.into(%{})
```
example:
```
 zhyu@zhyus-MacBook-Pro  ~/Workspace/elixir/nadia   master  iex -S mix
Erlang/OTP 23 [erts-11.0.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]

Interactive Elixir (1.10.3) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> user = %Nadia.Model.User{id: "fake_id"}
%Nadia.Model.User{first_name: nil, id: "fake_id", last_name: nil, username: nil}
iex(2)> user |> Map.from_struct() |> Enum.filter(fn {_, v} -> v != nil end) |> Enum.into(%{})
%{id: "fake_id"}
iex(3)>
```

File: mix.lock [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-  "makeup_elixir": {:hex, :makeup_elixir, "0.13.0", "be7a477997dcac2e48a9d695ec730b2d22418292675c75aa2d34ba0909dcdeda", [:mix], [{:makeup, "~> 0.8", [hex: :makeup, repo: "hexpm", optional: false]}], "hexpm"},
-  "meck": {:hex, :meck, "0.8.13", "ffedb39f99b0b99703b8601c6f17c7f76313ee12de6b646e671e3188401f7866", [:rebar3], [], "hexpm"},
-  "metrics": {:hex, :metrics, "1.0.1", "25f094dea2cda98213cecc3aeff09e940299d950904393b2a29d191c346a8486", [:rebar3], [], "hexpm"},
-  "mimerl": {:hex, :mimerl, "1.2.0", "67e2d3f571088d5cfd3e550c383094b47159f3eee8ffa08e64106cdf5e981be3", [:rebar3], [], "hexpm"},
-  "nimble_parsec": {:hex, :nimble_parsec, "0.5.0", "90e2eca3d0266e5c53f8fbe0079694740b9c91b6747f2b7e3c5d21966bba8300", [:mix], [], "hexpm"},
-  "parse_trans": {:hex, :parse_trans, "3.3.0", "09765507a3c7590a784615cfd421d101aec25098d50b89d7aa1d66646bc571c1", [:rebar3], [], "hexpm"},
+  "bunt": {:hex, :bunt, "0.2.0", "951c6e801e8b1d2cbe58ebbd3e616a869061ddadcc4863d0a2182541acae9a38", [:mix], [], "hexpm", "7af5c7e09fe1d40f76c8e4f9dd2be7cebd83909f31fee7cd0e9eadc567da8353"},
```
Comment (zhyu) [inline mix.lock:2]: I'm not sure where does the format change of mix.lock come. But seems your changes don't really need to change the lock file

Comment (Hentioe) [reply mix.lock:2]: I don't know what is going on about this. After I opened the project, ElixirLS automatically executed the command. Maybe the Mix version is inconsistent?

**Comment by Hentioe** [comment]
I understand what you mean by these, and I'll commit a fix later.

**Comment by Hentioe** [comment]
I improved the code. But sorry, I haven't used the exvcr library. I don't know how the json in the `fixture/vcr_cassettes` directory is obtained. Is it all handwritten? Or was it intercepted during the request?

**Comment by zhyu** [comment]
@Hentioe Yeah, VCR Cassettes were intercepted from requests. Tests can be written in this way:
1. Add a new test, e.g., `test "restrict_chat_member" do`
2. In the newly added test, call `restrict_chat_member` with a real chat id and other options
3. Run `mix test`, a new JSON file should be recorded
4. Make sure there's no sensitive data in the newly recorded JSON file, if there are any, you may need to update/add sensitive filters, check `setup` of `nadia_test.exs`
5. Replace sensitive data in test codes before committing

Please let me know if there are any issues during this process, I'm happy to help.

**Pull request closed without merging**